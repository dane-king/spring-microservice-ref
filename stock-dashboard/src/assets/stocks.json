[
  {
    "adp_100": "169.0224",
    "adp_200": "158.2149",
    "adp_50": "170.9848",
    "datetime": "2022-02-18T00:00:00-05:00",
    "last": "167.3000",
    "symbol": "AAPL",
    "vl": "82772674"
  },
  {
    "adp_100": "128.1137",
    "adp_200": "129.6569",
    "adp_50": "133.5882",
    "datetime": "2022-02-18T00:00:00-05:00",
    "last": "124.3500",
    "symbol": "IBM",
    "vl": "4609212"
  },
  {
    "adp_100": "5.8492",
    "adp_200": "5.8353",
    "adp_50": "5.8234",
    "datetime": "2022-02-18T00:00:00-05:00",
    "last": "5.6600",
    "symbol": "NOK",
    "vl": "26626039"
  },
  {
    "adp_100": "20.3985",
    "adp_200": "17.4677",
    "adp_50": "20.8271",
    "datetime": "2022-02-18T00:00:00-05:00",
    "last": "18.0400",
    "symbol": "F",
    "vl": "106404212"
  }
]
