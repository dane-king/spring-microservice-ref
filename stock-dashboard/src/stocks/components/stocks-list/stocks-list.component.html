<div id="stock-list">
  <h2>Stock List</h2>
  <table class="shadow-lg bg-white">
    <tr>
      <th class="bg-blue-100 border text-left px-8 py-4">Ticker</th>
      <th class="bg-blue-100 border text-left px-8 py-4">Name</th>
      <th class="bg-blue-100 border text-left px-8 py-4">Last</th>
      <th class="bg-blue-100 border text-left px-8 py-4">Amount Change</th>
      <th class="bg-blue-100 border text-left px-8 py-4">Percent Change</th>
    </tr>
    <tr *ngFor="let element of stocks">
      <td class="border px-8 py-4">
        <a [routerLink]="['/stocks', element.ticker]"> {{ element.ticker }} </a>
      </td>
      <td class="border px-8 py-4">{{ element.name }}</td>
      <td class="border px-8 py-4">{{ element.last }}</td>
      <td class="border px-8 py-4">{{ element.change }}</td>
      <td class="border px-8 py-4"> {{ element.percent_change | percent: ".2-4" }}</td>
    </tr>
  </table>
</div>
