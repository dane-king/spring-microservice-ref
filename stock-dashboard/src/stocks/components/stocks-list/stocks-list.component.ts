import { Component, OnInit } from '@angular/core';
import { Stock } from '../../model/stock';


const ELEMENT_DATA: Stock[] = [
  {ticker:'IBM', name: 'International Business Machine', last: 124.35, change: .62, percent_change: .01},
  {ticker:'AAPL', name: 'Apple Inc', last: 167.30, change: -1.58, percent_change: -.5},
  {ticker:'NOK', name: 'Nokia', last: 5.66, change: .01, percent_change: .01},
];


@Component({
  selector: 'app-stocks-list',
  templateUrl: './stocks-list.component.html',
  styleUrls: ['./stocks-list.component.scss']
})

export class StocksListComponent {
  displayedColumns: string[] = ['ticker', 'name', 'last', 'change', 'percent_change'];
  dataSource = ELEMENT_DATA;

  constructor() { }


}
