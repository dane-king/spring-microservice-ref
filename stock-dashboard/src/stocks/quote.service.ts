import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, of} from 'rxjs';
import { Stock } from './model/stock';

@Injectable({
  providedIn: 'root'
})
export class QuoteService {
  //TODO: use environment variable to get the data
  url = "localhost:3000/stock-quote/";

  private httpOptions = {
    headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
  };

  constructor(private httpClient:HttpClient) { }


  getAllQuotes(): Observable<Stock[]> {
    return this.httpClient.get<Stock[]>(`${this.url}/quotes`);
  }

}
