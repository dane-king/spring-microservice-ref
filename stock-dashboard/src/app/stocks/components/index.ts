import {StockCardComponent} from './stock-card/stock-card.component';
import {StocksListComponent} from './stocks-list/stocks-list.component';

export const components=[
  StockCardComponent,
  StocksListComponent
];

export * from './stock-card/stock-card.component';
export * from './stocks-list/stocks-list.component';
