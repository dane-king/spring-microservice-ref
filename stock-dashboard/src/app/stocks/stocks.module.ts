import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
// components
import * as fromComponents from './components';

import {MatCardModule} from '@angular/material/card';
import {MatTableModule} from '@angular/material/table';


@NgModule({
  declarations: [...fromComponents.components],
  imports: [
    CommonModule,
    MatCardModule,
    MatTableModule
  ],
  exports: [
    ...fromComponents.components,
  ]
})
export class StocksModule { }
