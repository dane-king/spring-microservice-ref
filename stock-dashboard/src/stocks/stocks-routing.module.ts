import { StocksListComponent } from './components/stocks-list/stocks-list.component';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { components, StockCardComponent } from './components';
const routes: Routes = [
  {
    path: "",
    component: StocksListComponent
  },
  { path: ':id', component: StockCardComponent },

];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class StocksRoutingModule { }
