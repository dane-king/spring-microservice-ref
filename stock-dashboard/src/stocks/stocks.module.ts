import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { StocksRoutingModule } from './stocks-routing.module';

// components
import * as fromComponents from './components';


@NgModule({
  declarations: [...fromComponents.components],
  imports: [
    CommonModule,
    StocksRoutingModule
  ],
  exports: [
    ...fromComponents.components,
  ]
})
export class StocksModule { }
