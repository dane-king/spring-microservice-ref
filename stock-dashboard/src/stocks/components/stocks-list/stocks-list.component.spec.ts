import { routes } from './../../../app/app-routing.module';
import { StocksModule } from './../../stocks.module';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { StocksListComponent } from './stocks-list.component';
import { Router, Routes } from '@angular/router';

describe('StocksListComponent', () => {
  let component: StocksListComponent;
  let fixture: ComponentFixture<StocksListComponent>;
  let htmlBody:HTMLElement;
  let router:Router;
  let location:Location;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ StocksListComponent],
      imports:[StocksModule, RouterTestingModule.withRoutes(routes)]
    })
    .compileComponents();

  });

  beforeEach(() => {
    fixture = TestBed.createComponent(StocksListComponent);
    component = fixture.componentInstance;
    htmlBody = fixture.debugElement.nativeElement;
    fixture.detectChanges();
  });

  it('should have render properly', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should have list of items', ()=>{
    expect(htmlBody.querySelector('#stock-list')).toBeTruthy();
  })
  it('should go to home if link is invalid', () => {

  });
  it('should go to stock card from ticker link', () => {

  });
});
function routes(routes: any): any {
  throw new Error('Function not implemented.');
}

