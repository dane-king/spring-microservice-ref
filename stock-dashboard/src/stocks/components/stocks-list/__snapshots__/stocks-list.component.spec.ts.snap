// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StocksListComponent should have render properly 1`] = `
<app-stocks-list
  __ngContext__={[Function LRootView]}
  displayedColumns={[Function Array]}
  stocks={[Function Array]}
>
  <div
    id="stock-list"
  >
    <h2>
      Stock List
    </h2>
    <table
      class="shadow-lg bg-white"
    >
      <tr>
        <th
          class="bg-blue-100 border text-left px-8 py-4"
        >
          Ticker
        </th>
        <th
          class="bg-blue-100 border text-left px-8 py-4"
        >
          Name
        </th>
        <th
          class="bg-blue-100 border text-left px-8 py-4"
        >
          Last
        </th>
        <th
          class="bg-blue-100 border text-left px-8 py-4"
        >
          Amount Change
        </th>
        <th
          class="bg-blue-100 border text-left px-8 py-4"
        >
          Percent Change
        </th>
      </tr>
      <tr>
        <td
          class="border px-8 py-4"
        >
          <a
            href="/stocks/IBM"
          >
             IBM 
          </a>
        </td>
        <td
          class="border px-8 py-4"
        >
          International Business Machine
        </td>
        <td
          class="border px-8 py-4"
        >
          124.35
        </td>
        <td
          class="border px-8 py-4"
        >
          0.62
        </td>
        <td
          class="border px-8 py-4"
        >
           1.00%
        </td>
      </tr>
      <tr>
        <td
          class="border px-8 py-4"
        >
          <a
            href="/stocks/AAPL"
          >
             AAPL 
          </a>
        </td>
        <td
          class="border px-8 py-4"
        >
          Apple Inc
        </td>
        <td
          class="border px-8 py-4"
        >
          167.3
        </td>
        <td
          class="border px-8 py-4"
        >
          -1.58
        </td>
        <td
          class="border px-8 py-4"
        >
           -50.00%
        </td>
      </tr>
      <tr>
        <td
          class="border px-8 py-4"
        >
          <a
            href="/stocks/NOK"
          >
             NOK 
          </a>
        </td>
        <td
          class="border px-8 py-4"
        >
          Nokia
        </td>
        <td
          class="border px-8 py-4"
        >
          5.66
        </td>
        <td
          class="border px-8 py-4"
        >
          0.01
        </td>
        <td
          class="border px-8 py-4"
        >
           1.00%
        </td>
      </tr>
    </table>
  </div>
</app-stocks-list>
`;
